packages/contractkit/src/identity/claims/verify.ts

```diff
@@ -70,37 +69,45 @@ type dnsResolverFunction = (
  * `celo-site-verification` and a valid signature in base64
  */
 export const verifyDomainRecord = async (
+  kit: ContractKit,
   claim: DomainClaim,
   address: string,
   dnsResolver: dnsResolverFunction = resolveTxt as any
 ) => {
   const found = await new Promise((resolve) => {
-    dnsResolver(claim.domain, (error, domainRecords) => {
+    dnsResolver(claim.domain, async (error, domainRecords) => {
       if (error) {
         console.debug(`Unable to fetch domain TXT records: ${error.toString()}`)
-        resolve(false)
+        resolve(`Unable to fetch domain TXT records: ${error.toString()}`)
```

Comment:
We should be using `reject` here. Resolving truthy values that are meant to be a failures can lead to misinterpreted results in consumers.

Reply:
Agree, but a different pattern is used here for some reason (undefined means success and returned strings are error messages). Anyway cleaned it a bit up to make it easier to understand.